pilot/pkg/model/push_context.go

```diff
@@ -496,13 +496,6 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 
 	// FIXME: this code should be removed once the EDS issue is fixed
 	if proxy == nil {
-		// look for dest rules across all namespaces public/private
```

Comment:
If we could remove that logic completely, that would be much better. I don't know the why do we have these code in first place. As long as the change doesn't break anything existing, we should go for it.

@rshriram any thoughts?

Reply:
I dont know how much changes have happened here, but this code exists soley for EDS.. EDS is still not updated to use the scopes and exports.. So it calls ps.SubsetToLabels(nil, ..) which in turn calls DestinationRule with nil value for the proxy..
```
 // TODO: BUG. this code is incorrect if 1.1 isolation is used. With destination rule scoping
        // (public/private) as well as sidecar scopes allowing import of
        // specific destination rules, the destination rule for a given
        // namespace should be determined based on the sidecar scope or the
        // proxy's config namespace. As such, this code searches through all
        // destination rules, public and private and returns a completely
        // arbitrary destination rule's subset labels!
        labels := push.SubsetToLabels(nil, subsetName, hostname)
```

This is in @costinm 's territory..

Reply:
IMO the change to EDS is too risky for 1.1. We can fix it in 1.1.x

Removing code - after the new code is proven to be stable and work, of course. Normal process is to have new features opt-in, and after it is used with success in a release we can make it default and remove the old one.

Reply:
I don't think this is removing code that is proven to be stable and work. This code was new in 1.1.

How this code works is to find the first host that matches, by going through each namespace in arbitrary order, then by the Hostname ordering. This is what is causing the bug - we pick istio-system first, match to the default rule, and get the wrong settings. By removing the code, we go down to the next block (currently never executed, as we will always find a match before this), which goes through each one, in Hostname ordering. By doing this we will ensure we pick the more specific rule instead of the default rule. I am not sure what the issue is with removing this code?